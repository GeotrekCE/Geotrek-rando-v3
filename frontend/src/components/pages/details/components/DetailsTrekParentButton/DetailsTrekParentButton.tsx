import { ArrowLeft } from 'components/Icons/ArrowLeft';
import { Link } from 'components/Link';
import { generateResultDetailsUrl } from 'components/pages/search/utils';

interface DetailsTrekParentButtonProps {
  parentName: string;
  parentId: string;
  className?: string;
}

export const DetailsTrekParentButton: React.FC<DetailsTrekParentButtonProps> = ({
  parentName,
  parentId,
  className,
}) => {
  return (
    <Link href={generateResultDetailsUrl(parentId, parentName)} className={className}>
      <span className="flex items-center">
        <ArrowLeft size={20} className="mr-1" />
        <span className="underline text-Mobile-C2 desktop:text-P1">{parentName}</span>
      </span>
    </Link>
  );
};
