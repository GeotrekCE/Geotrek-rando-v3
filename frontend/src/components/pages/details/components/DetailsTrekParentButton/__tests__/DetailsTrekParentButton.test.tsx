import { render } from 'services/testing/reactTestingLibraryWrapper';
import { DetailsTrekParentButton } from '../DetailsTrekParentButton';
describe('DetailsTrekParentButton', () => {
  it('should display a DetailsTrekParentButton', () => {
    const component = render(
      <DetailsTrekParentButton parentId={'1'} parentName="Test rando mÃ¨re" />,
    );

    expect(component).toMatchSnapshot();
  });
});
