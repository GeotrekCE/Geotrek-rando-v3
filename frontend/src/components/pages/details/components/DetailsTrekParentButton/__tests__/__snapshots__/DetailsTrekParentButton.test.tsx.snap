// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsTrekParentButton should display a DetailsTrekParentButton 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="sc-bdfBQB btWAAI"
        href="/trek/1-Test-rando-mere"
      >
        <div
          class="flex items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 12H5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M12 19L5 12L12 5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span
            class="underline text-Mobile-C2 desktop:text-P1"
          >
            Test rando mère
          </span>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="sc-bdfBQB btWAAI"
      href="/trek/1-Test-rando-mere"
    >
      <div
        class="flex items-center"
      >
        <svg
          class="mr-1"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 12H5"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M12 19L5 12L12 5"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <span
          class="underline text-Mobile-C2 desktop:text-P1"
        >
          Test rando mère
        </span>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
